app/src/main/java/org/oppia/android/app/drawer/ExitProfileDialogFragment.kt

```diff
@@ -53,7 +72,11 @@ class ExitProfileDialogFragment : DialogFragment() {
       .setMessage(R.string.home_activity_back_dialog_message)
       .setNegativeButton(R.string.home_activity_back_dialog_cancel) { dialog, _ ->
         if (isFromNavigationDrawer) {
-          exitProfileDialogInterface.markHomeMenuCloseDrawer()
```

Comment:
@anandwana001 I commented out ```exitProfileDialogInterface.markHomeMenuCloseDrawer()```, so after this ```Switch Profile``` was getting marked always, this proves that the bug originated from here. What do you say?

Reply:
it's actually happening before this code arrives too. 
I think we need to check on how the menu item selection is occurring and how are we handling the previous item thing.